[DC-796] pylint: prefer fstring syntax for log statements

calbach opened the pull request:
@ksdkalluri FYI

lrwb-aou commented on .pylintrc:
```diff
@@ -232,7 +232,7 @@ never-returning-functions=sys.exit
 # Format style used to check logging format string. `old` means using %
 # formatting, `new` is for `{}` formatting,and `fstr` is for f-strings.
-logging-format-style=old
+logging-format-style=fstr
```
This only works for pylint version 2.4.X.  It is immediately broken in pylint version 2.5.  I'm putting this here for documentation.
 https://github.com/PyCQA/pylint/issues/3570#issuecomment-622980946

calbach commented:
Fixed all resulting pylint warnings. This certainly blew up a bit in scope. I fixed most of them with a nasty emacs `regexp-replace`: 

```
\(\(LOGGER\|logging\).[a-z]+(\)[ ^J]*f?["']\([^'"%]+\)%[sdf]\([^'"]*\)\
['"],[ ^J]*\([^,)]+\) → \1f"\3{\5}\4"
```

the rest I fixed up by hand.

calbach commented:
Please take another look when you have a chance.

lrwb-aou commented on data_steward/admin/key_rotation.py:
```diff
@@ -82,7 +82,7 @@ def delete_key(key):
         LOGGER.info(
             '{full_key_name} is deleted'.format(full_key_name=full_key_name))
```
shouldn't this also be an f-string?

calbach replied:
Indeed.. pylint appears to not care about `format` style logs. In some cases I've found it actually may still be necessary to use this style - specifically because you can't inline backslashes in an f-string expression (see further down in the PR). We could perhaps revisit when upgrading pylint. Fixed for now though

lrwb-aou commented on data_steward/deid/aou.py:
```diff
@@ -666,8 +668,8 @@ def submit(self, sql, create, dml=None):
                 LOGGER.info(
-                    'submitted a %s job for table:\t%s\t\tstatus:\t%s\t\tvalue:\t%s',
-                    'bigquery', table_name, 'pending', response.job_id)
+                    f"submitted a bigquery job for table:\t{table_name}\t\tstatus:\t'pending'\t\tvalue:\t{response.job_id}"
```
nit:  runaway line length.

calbach replied:
fixed

lrwb-aou commented on data_steward/deid/press.py:
```diff
@@ -323,8 +324,8 @@ def simulate(self, info):
                 LOGGER.info(
-                    'no data-found for simulation of table:\t%s\t\tfield:\t%s\t\ttype:\t%s',
-                    table_name, field, item['label'])
+                    f"no data-found for simulation of table:\t{table_name}\t\tfield:\t{field}\t\ttype:\t{item['label']}"
```
nit:  runaway line length.

calbach replied:
done

lrwb-aou commented on data_steward/tools/run_deid.py:
```diff
@@ -252,26 +252,27 @@ def main(raw_args=None):
-    LOGGER.info('Deid has finished.  Successfully executed on tables:  %s',
-                '\n'.join(successes))
+    LOGGER.info(
+        "Deid has finished.  Successfully executed on tables: {}".format(
+            '\n'.join(successes)))
```
nit:  weird.  this changed to a formatted string instead of an f-string.

calbach replied:
Per above comment, this case caused me to discover a limitation of fstrings I wasn't aware of. Backslashes cannot be embedded into fstring expressions. My options here were:

- Use a format or substitution
- assign to a variable then fstring interpolate the variable

There's an argument for the latter instead, but I just went with this since apparently `.format` is still legal in this world. Let me know if you prefer the latter approach

lrwb-aou replied:
I think the `.format` is fine.  Adding this to a variable to allow f-string interpolation feels like too much extra work just for a logging statement.  I didn't realize there was a limitation with backslashes in f-strings either.  I just noticed the oddity and wondered why it existed.

lrwb-aou commented on data_steward/validation/main.py:
```diff
@@ -351,9 +351,8 @@ def generate_metrics(hpo_id, bucket, folder_prefix, summary):
-            'Processing complete. Saving timestamp %s to `gs://%s/%s`.',
-            processed_datetime_str, bucket,
-            folder_prefix + common.PROCESSED_TXT)
+            f"Processing complete. Saving timestamp {processed_datetime_str} to `gs://{bucket}/{folder_prefix + common.PROCESSED_TXT}`."
```
nit:  runaway line length.

calbach replied:
done

lrwb-aou commented on data_steward/validation/main.py:
```diff
@@ -398,10 +397,9 @@ def generate_empty_report(hpo_id, bucket, folder_prefix):
-        'Saving timestamp %s to `gs://%s/%s`.', folder_prefix,
-        consts.FOLDER_NAMING_CONVENTION, processed_datetime_str, bucket,
-        folder_prefix + common.PROCESSED_TXT)
+        f"Processing skipped. Reason: Folder {folder_prefix} does not follow naming convention {consts.FOLDER_NAMING_CONVENTION}. "
+        f"Saving timestamp {processed_datetime_str} to `gs://{bucket}/{folder_prefix + common.PROCESSED_TXT}`."
```
nit:  runaway line length.

calbach replied:
done

calbach commented on data_steward/cdr_cleaner/clean_cdr_engine.py:
```diff
@@ -74,9 +74,9 @@ def clean_dataset(project=None, statements=None, data_stage=stage.UNSPECIFIED):
-                        index, statement_length, module_name, function_name,
-                        rule_query)
+            LOGGER.info(
+                f"Running query {index} out of {statement_length} generated by {module_name}.{function_name} \n{rule_query}"
```
Ah I see.. thanks. I don't think I understood the rationale for pylint, but I see it now.
